onnxruntime/core/providers/dml/DmlExecutionProvider/src/DmlGraphFusionTransformer.cpp

```diff
@@ -0,0 +1,391 @@
+#pragma once
+
+#include "precomp.h"
+#include "GraphDescBuilder.h"
+#include "ExecutionProvider.h"
+#include "DmlGraphFusionTransformer.h"
+#include "GraphPartitioner.h"
+#include "core/framework/kernel_type_str_resolver.h"
+#include "core/framework/kernel_lookup.h"
+#include "FusedGraphKernel.h"
+#include "MLOperatorAuthorImpl.h"
+#include "GraphKernelHelper.h"
+
+
+namespace Dml
+{
+    DmlGraphFusionTransformer::DmlGraphFusionTransformer(
+        const std::string& name,
+        const onnxruntime::IExecutionProvider* provider
+    )
+        :onnxruntime::GraphTransformer(name),
+         m_providerImpl(static_cast<const ExecutionProvider*>(provider)->GetImpl())
+    {
+    }
+	
+    std::unordered_map<const onnx::TensorProto*, std::vector<uint32_t>>
+    GetInitializerToPartitionMap(
+        const onnxruntime::GraphViewer& graph,
+        gsl::span<std::unique_ptr<GraphPartition>> partitions
+    )
+    {
+        std::unordered_map<const onnx::TensorProto*, std::vector<uint32_t>> initializerPartitionMap;
+        for (uint32_t partitionIndex = 0; partitionIndex < gsl::narrow_cast<uint32_t>(partitions.size()); ++partitionIndex)
+        {
+            auto& partition = partitions[partitionIndex];
+
+            // Skip partitions which have been merged into other partitions
+            if (partition->GetRootMergedPartition() != partition.get())
+            {
+                continue;
+            }
+
+            for (const std::string& input : partition->GetInputs())
+            {
+                const onnx::TensorProto* tensor = nullptr;
+                if (graph.GetInitializedTensor(input, tensor))
+                {
+                    initializerPartitionMap[tensor].push_back(partitionIndex);
+                }
+            }
+        }
+
+        return initializerPartitionMap;
+    }
+
+    void CreateONNXGraphForPartition(
```

Comment:
>CreateONNXGraphForPartition [](http://example.com/codeflow?start=9&length=27)

This seems to make extra copies of the sub graph and weights.  Because the resulting partitionGraph is just used by the same transform in BuildGraphDesc, can that code be adapted to read the original graph structure instead?

Reply:
Looks like it can be. Did that.